assets/js/event-providers/utils.js

```diff
@@ -0,0 +1,229 @@
+	if ( normalizedPhone.startsWith( '+' ) ) {
+		return `+${ digits }`;
+	}
+
+	return digits;
+}
+
+/**
+ * Checks if a value is likely an email address.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} value The value to check.
+ * @return {boolean} True if the value is likely an email address, false otherwise.
+ */
+export function isLikelyEmail( value ) {
+	const normalizedEmail = normalizeEmail( value );
+
+	const emailPattern = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+
+	return emailPattern.test( normalizedEmail );
+}
+
+/**
+ * Checks if a value is likely a phone number.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} value The value to check.
+ * @return {boolean} True if the value is likely a phone number, false otherwise.
+ */
+export function isLikelyPhone( value ) {
+	const normalizedPhone = normalizePhone( value );
+
+	const phonePattern = /^\+?\d{7,}$/;
+
+	return phonePattern.test( normalizedPhone );
+}
+
+/**
+ * Classifies a field as PII based on its metadata.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object} fieldMeta The metadata of the field to classify.
+ * @return {Object} An object containing the PII type and normalized value.
+ */
+export function classifyPII( fieldMeta ) {
+	let { type, name, value, label } = fieldMeta;
+
+	type = normalizeValue( type );
+	name = normalizeValue( name );
+	value = normalizeValue( value );
+	label = normalizeValue( label );
+
+	switch ( type ) {
+		case 'email':
+			return {
+				type: PII_TYPE.EMAIL,
+				value: normalizeEmail( value ),
+			};
+		case 'tel':
+			return {
+				type: PII_TYPE.PHONE,
+				value: normalizePhone( value ),
+			};
+	}
+
+	if ( isLikelyEmail( value ) ) {
+		return {
+			type: PII_TYPE.EMAIL,
+			value: normalizeEmail( value ),
+		};
+	}
+
+	if ( isLikelyPhone( value ) ) {
+		return {
+			type: PII_TYPE.PHONE,
+			value: normalizePhone( value ),
+		};
+	}
+
+	const indicators = {
+		[ PII_TYPE.EMAIL ]: [ 'email', 'e-mail', 'mail', 'email address' ],
+		[ PII_TYPE.PHONE ]: [
+			'phone',
+			'tel',
+			'mobile',
+			'cell',
+			'telephone',
+			'phone number',
+		],
+		[ PII_TYPE.NAME ]: [
+			'name',
+			'full-name',
+			'full name',
+			'fullname',
+			'first-name',
+			'first name',
+			'firstname',
+			'last-name',
+			'last name',
+			'lastname',
+			'given-name',
+			'given name',
+			'givenname',
+			'family-name',
+			'family name',
+			'familyname',
+			'fname',
+			'lname',
+		],
+	};
+
+	if (
+		indicators[ PII_TYPE.EMAIL ].includes( name ) ||
+		indicators[ PII_TYPE.EMAIL ].includes( label )
+	) {
+		return {
+			type: PII_TYPE.EMAIL,
+			value: normalizeEmail( value ),
+		};
+	}
+
+	if (
+		indicators[ PII_TYPE.PHONE ].includes( name ) ||
+		indicators[ PII_TYPE.PHONE ].includes( label )
+	) {
+		return {
+			type: PII_TYPE.PHONE,
+			value: normalizePhone( value ),
+		};
+	}
+
+	if (
+		indicators[ PII_TYPE.NAME ].includes( name ) ||
+		indicators[ PII_TYPE.NAME ].includes( label )
+	) {
+		return {
+			type: PII_TYPE.NAME,
+			value: normalizeValue( value ),
+		};
+	}
```

Comment:
Explicit `null` return ensures consistent behavior.
```suggestion
	}
	
	return null;
```